src/daemon/collectd.c

```diff
@@ -86,13 +67,9 @@ static int init_hostname(void) {
     return 0;
   }
 
-  long hostname_len = sysconf(_SC_HOST_NAME_MAX);
-  if (hostname_len == -1) {
-    hostname_len = NI_MAXHOST;
-  }
-  char hostname[hostname_len];
+  char hostname[hostname_len()];
```

Comment:
Why did you introduce hostname_len() and call it twice? I think the code can continue to be inlined here.

Reply:
This was to separate out some Windows vs. other OS behavior from the main logic of the program, but probably doesn't belong in this PR since it isn't actually needed yet.